config/rbac/role_database.yaml

```diff
@@ -0,0 +1,20 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: mongodb-database
+  namespace: mongodb
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    verbs:
+      - get
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - patch
+      - delete
```

Comment:
Q: I don't recall why it needs the delete permission?

Reply:
got it, @chatton do you happen to know why we have the delete permission -- can we delete it from both the places ?

Reply:
@irajdeep https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/cmd/versionhook/main.go#L197 Here's the incriminated line :)

Reply:
got it -- thanks.